cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1358,6 +1369,12 @@ def _add_head_node(self):
                         else "false"
                     ),
                     "launch_template_id": launch_template_id,
+                    **(
+                        {"p6egb200_block_sizes": cluster_ultraserver_capacity_block_sizes_dict["p6e-gb200"]}
+                        if "p6e-gb200" in cluster_ultraserver_capacity_block_sizes_dict
+                        and cluster_ultraserver_capacity_block_sizes_dict["p6e-gb200"]
```

Comment:
Can you add unit tests for checking this is generated as expected?
It would also be easier to understand the structure using unit tests